pkgs/os-specific/linux/switcheroo-control/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, ninja, meson, fetchFromGitLab, systemd, libgudev, pkg-config, glib, python3Packages, gobject-introspection }:
+python3Packages.buildPythonApplication rec {
+  pname = "switcheroo-control";
+  version = "2.3";
+  
+  format = "other";
+  src = fetchFromGitLab {
+    hash = "sha256-1Pze2TJ9mggfcpiLFwJ7/9WhsdJx4G3GoA7+Z47shuc=";
+    domain = "gitlab.freedesktop.org";
+    owner = "hadess";
+    repo = pname;
+    rev = version;
+  };
+  
+  nativeBuildInputs = [
+    ninja meson pkg-config glib
+  ];
+  buildInputs = [
+    systemd libgudev
+  ];
+  propagatedBuildInputs = [ python3Packages.pygobject3 ];
+  
+  mesonFlags = [ 
+    "-Dsystemdsystemunitdir=${placeholder "out"}/etc/systemd/system" 
+    "-Dhwdbdir=${placeholder "out"}/etc/udev/hwdb.d"
+  ];
+  
+  meta = with lib; {
+    description = "D-Bus service to check the availability of dual-GPU";
+    homepage = "https://gitlab.freedesktop.org/hadess/switcheroo-control/";
+    changelog = "https://gitlab.freedesktop.org/hadess/switcheroo-control/-/blob/${version}/NEWS";
+    license = licenses.gpl3Only;
```

Comment:
Hmm, it is inconsistent https://gitlab.freedesktop.org/hadess/switcheroo-control/-/issues/18